### PR TITLE
chore(ifo): Support deposit

### PR DESCRIPTION
**Pull request by apt-waffle**
- Add more structs as interfaces.
- Add Move view functions as TypeScript functions.
- Implement functionality for deposit.
- Remove manual `fetchIfoData` since we poll via `useQuery` for Aptos. This fixes some FE sync issues when fetching from multiple tables.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
     "next-themes": "^0.2.0",
     "numeral": "^2.0.6",
     "react": "^18.2.0",
-    "react-countup": "^6.3.2",
+    "react-countup": "^6.3.1",
```
Comment (chef-jojo) [inline package.json:90]: Why downgraded?

Comment (apt-waffle) [reply package.json]: Accidental sorry, fixed!

File: apps/aptos/views/Ifos/types.ts [hunk 2]
```diff
@@ -62,21 +72,11 @@ export interface WalletIfoState {
...
-     * credit left is the ifo credit minus the amount of `amountTokenCommittedInLP` in unlimited pool
-     * minimum is 0
-     */
-    creditLeft: BigNumber
-  }
 }
 
 // Returned by useGetWalletIfoData
 export interface WalletIfoData extends WalletIfoState {
   contract: Contract // TODO: Aptos
```
Comment (chef-jojo) [inline apps/aptos/views/Ifos/types.ts:79]: can remove?

File: apps/aptos/views/Ifos/hooks/vesting/useFetchUserWalletIfoData.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  userVestingData: {
+    vestingStartTime: number
+    [PoolIds.poolUnlimited]: VestingCharacteristics
+  }
+}
+
+export const useFetchUserWalletIfoData = () => {
+  const resources = useIfoResources()
+  const pool = useIfoPool()
+  const vestingCharacteristics = useVestingCharacteristics()
```
Comment (chef-jojo) [inline apps/aptos/views/Ifos/hooks/vesting/useFetchUserWalletIfoData.ts:23]: seems like we need to refetch these three hooks on `fetchUserWalletIfoData`
can get the `refetch` and await on below callback
`const { data: pool, refetch: refetchPool } = useIfoPool();`

**Pull request closed without merging**